### PR TITLE
Try to fix OS X build

### PR DESCRIPTION
**Pull request by rcurtin**
Another day, another strange issue that has nothing to do with mlpack.  Sigh.

(Let's see how many iterations it takes to fix.)

### CODE REVIEW

**Comment by rcurtin** [comment]
@mlpack-jenkins test this please

**Comment by rcurtin** [comment]
The R build is still failing due to the deprecation issues, but the PR itself does its job (fix the regular OS X build) so it's ready for review.

**Comment by eddelbuettel** [comment]
Did you see the similar hotfix in the r-actions thread I mentioned on Matrix? In [this comment](https://github.com/r-lib/actions/issues/948#issuecomment-2495861299) Kirill suggests to uninstall as you do but then (as I read it) let the R install take care of the remainder.

Anyway, it seems as if the runner image will come back repaired in due course too.

**Comment by rcurtin** [comment]
Yeah, I saw that, but for the jobs I fixed here, they are the general mlpack OS X jobs, so it is not guaranteed that R will even be used.  So in that case it makes sense just to reinstall pkg-config on the spot.

**Comment by eddelbuettel** [comment]
Do you want me to approve this?

**Comment by rcurtin** [comment]
If you're comfortable with the change, then yes please :)

File: .ci/macos-steps.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ steps:
 - script: |
     set -e
     sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
+    brew uninstall pkg-config@0.29.2
+    brew install pkg-config
```
Comment (eddelbuettel) [inline .ci/macos-steps.yaml:17]: As this is a hot-fix for a temporary issue, shall we place a 'reminder' somewhere to see about reverting to normal "in due course" ?

**Review by eddelbuettel — APPROVED** [review]
One comment inline

File: .ci/macos-steps.yaml (hunk 1 above)
Comment (rcurtin) [reply .ci/macos-steps.yaml:19]: Good idea, done in aeac96e0d4. :+1:

**Comment by rcurtin** [comment]
Well, I looked on another PR and everything is working there.  So apparently this was a transient problem that doesn't even need a workaround anymore.

**Pull request closed without merging**